pkg/catalog/manager/catalog_sync.go

```diff
@@ -43,13 +47,54 @@ func (m *Manager) Sync(key string, obj *v3.Catalog) (runtime.Object, error) {
 	}
 	logrus.Debugf("Chart hash comparison for global catalog %v: new -- %v --- current -- %v", catalog.Name, commit, catalog.Status.Commit)
 
+	// ensure the system catalog image cache exists
+	var systemCatalogImageCache *v1.ConfigMap
+	if catalog.Name == utils.SystemLibraryName {
+		systemCatalogImageCacheName := utils.GetCatalogImageCacheName(catalog.Name)
+		systemCatalogImageCache, err = m.ConfigMapLister.Get(systemCatalogImageCacheName, metav1.GetOptions{})
+		if err != nil {
+			if apiErr, ok := err.(*errors.StatusError); !ok || apiErr.Status().Code != 404 {
+				return nil, err
+			}
+		}
+
+		// if the cache does not exist generate it
+		if systemCatalogImageCache.UID == "" {
```

Comment:
Better to check if error is not found then create, in case systemCatalogImageCache is a nil pointer